PlayCover/Views/Sidebar Views/AppFolderView.swift

```diff
@@ -0,0 +1,272 @@
+                                        showWrongfileTypeAlert = true
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+                return true
+            } else {
+                return false
+            }
+        }
+        .alert(isPresented: $showWrongfileTypeAlert) {
+            Alert(title: Text("alert.wrongFileType.title"),
+                  message: Text("alert.wrongFileType.subtitle"), dismissButton: .default(Text("button.OK")))
+        }
+        .alert("Legacy App Settings Detected!", isPresented: $showLegacyConvertAlert, actions: {
+            Button("button.Convert", role: .destructive) {
+                LegacySettings.convertLegacyMonolithPlist(LegacySettings.monolithURL)
+                do {
+                    try FileManager.default.removeItem(at: LegacySettings.monolithURL)
+                } catch {
+                    Log.shared.error(error)
+                }
+            }
+            .keyboardShortcut(.defaultAction)
+            Button("button.Cancel", role: .cancel) {
+                showLegacyConvertAlert.toggle()
+            }
+        }, message: {
+            Text("alert.legacyImport.subtitle")
+        })
+    }
+
+    private func installApp(_ url: URL) {
+        Installer.install(ipaUrl: url, export: false, returnCompletion: { _ in
+            Task { @MainActor in
+                appsVM.fetchApps()
+                NotifyService.shared.notify(
+                    NSLocalizedString("notification.appInstalled", comment: ""),
+                    NSLocalizedString("notification.appInstalled.message", comment: ""))
+            }
+        })
+    }
+
+    private func selectFile() {
+        NSOpenPanel.selectIPA { result in
+            if case .success(let url) = result {
+                installApp(url)
+            }
+        }
+    }
+}
+
+struct AddAppSheet: View {
+    @State var addSheetApps = false
+    @State var isAppEnabled: Bool
+    @State var app: PlayApp
+    @Binding var appList: Folder
+    var body: some View {
+        HStack {
+           // let image:NSImage = DataCache.instance.readImage(forKey: app.info.bundleIdentifier)
+            if let image = DataCache.instance.readImage(forKey: app.info.bundleIdentifier) {
+                Image(nsImage: image)
+                    .resizable()
+                    .cornerRadius(10)
+                    .shadow(radius: 1)
+                    .frame(width: 28, height: 28)
+            }
+            Toggle(app.info.displayName, isOn: $isAppEnabled)
+                .onChange(of: isAppEnabled) { _ in
+                    if isAppEnabled {
+                        appList.apps.append(app.info.bundleIdentifier)
+                    } else {
+                        appList.apps = appList.apps.filter { $0 != app.info.bundleIdentifier }
+                    }
+            }
+        }
+    }
+}
+
+let icons = [
```

Comment:
Best to put this inside a class or structure to avoid polluting the global scope.